File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;

class Solution {
    boolean solution(String s) {
        Stack<Character> stack = new Stack<>();
        
        for(char c : s.toCharArray()){
            if(c == '(') stack.push(c);
            else{
                if(stack.isEmpty()) return false;
                if(!stack.isEmpty() && stack.peek() == '(') stack.pop();
            }
        }
        
        if(stack.size() > 0) return false;
        return true;
    }
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []
    
    for c in s :
        if(c == ')') :
            if (len(stack) == 0) : return False
            elif (len(stack) != 0) : 
                if(stack.pop() == ')') :
                    return False
            
        else :
            stack.append(c)
        

    return len(stack) == 0 
